fast_plotter/plotting.py

```diff
@@ -191,6 +206,41 @@ def actually_plot(df, x_axis, y, yerr, kind, label, ax, dataset_col="dataset",
                           dataset_order=dataset_order,
                           line=False, expected_xs=expected_xs)
         vals.iloc[:, ::-1].apply(filler, axis=0, step="mid")
+    elif kind == "other_dtypes":
+        dtype_args = other_dtype_args[dtype]
+        dataset_colours =  dtype_args["colours"] if dtype_args["colours"] else dataset_colours
+        options = ["alpha", "style", "width", "add_label", "add_error", "regex"]
+        alpha, style, width, add_label, add_error, regex = [dtype_args[key] for key in options]
+        if "annotations" in dtype:
+            values = dtype_args['values'] if dtype_args['values'] else []
+            annotType = dtype_args["annotType"] if  dtype_args["annotType"] else "hlines"
+            z_order = dtype_args["z_order"]
+            if annotType == "hlines": 
+                xmin, xmax = [dtype_args[key] for key in ['xmin', 'xmax']]
+                xmax = len(expected_xs) if not xmax else xmax
+                plt.hlines(values, xmin, xmax, colors=dataset_colours,
+                           alpha=alpha, ls=style, lw = width, zorder=z_order)
+            if annotType == "vlines":
+                ymin = dtype_args['ymin']
+                ymax = dtype_args['ymax'] if dtype_args['ymax'] else 1
+                for value in values:
+                    plt.axvline(value, ymin, ymax, color=dataset_colours, alpha=alpha,
+                               ls=style, lw = width, zorder=z_order)
+        else:
+            filler = FillColl(n_datasets, ax=ax, fill=True, colourmap=colourmap, dataset_colours=dataset_colours,
+                          dataset_order=dataset_order, expected_xs=expected_xs, other_dtypes=other_dtype_args, 
+                          add_label_other=add_label, style_other=style, colour_other=dataset_colours)
+            vals.apply(filler, axis=0, step="mid")
+        for dset in list(set(df.reset_index()[dataset_col])):
```

Comment:
This bit is especially hacky. It's going through each input dataset that matches the regex and plotting the error for them individually. I haven't been able to get this to work unless the dataset colours are specified in the config. I'm sure there's a much more elegant way of doing this (and probably most of my changes but this especially jumps out)